pkg/utils/nodeclaim/nodeclaim.go

```diff
@@ -89,6 +89,26 @@ func NodePoolEventHandler(c client.Client) handler.EventHandler {
 	})
 }
 
+// NodeClassEventHandler is a watcher on v1beta1.NodeClaim that maps NodeClass to NodeClaims based
+// on the nodeClassRef and enqueues reconcile.Requests for the NodeClaim
+func NodeClassEventHandler(c client.Client) handler.EventHandler {
```

Comment:
Do we think that there's a way to test any of this logic? May be kind of tough to do in functional testing but we should think about some E2E test case that we can do to ensure that we are requeueing immediately on these changes.

If possible we can think about a way to leverage the `handler.EnqueueRequestsFromMapFunc` to try and conjure up some kind of functional test. I'm not sure how tough that it would be to do so, though

Reply:
For E2Es I was thinking we could have really short SLA on when we see the NodeClaims will get drifted on changes to both NodePool and NodeClass.